### ISSUE TITLE
Allow both HTTP and HTTPS

### ISSUE THREAD

**Issue by koenvervloesem** [rhasspy/rhasspy#28]
I have a setup with Node-RED and Rhasspy on the same Raspberry Pi, each in their own Docker container. I'd like to let Node-RED and Rhasspy communicate over unencrypted HTTP (WebSockets) because they're on the same machine anyway, while I use Rhasspy's web interface over HTTPS from my laptop.

Currently if I configure Rhasspy to use HTTPS, Node-RED can't communicate with Rhasspy over HTTP on Docker's internal network.

A solution would be that I could configure Rhasspy to:

- listen on HTTP on port X
- listen on HTTPS on port Y

And then I only expose port Y in the Docker container to outside, while I still can connect to port X from other containers on the same machine.

**Comment by synesthesiam** [comment]
I think this would be a good idea. I need to dig more into the `quart` library to see if it's possible out of the box, or if I'd need to do something extra. I'm guessing multiple versions of the quart app will need to be run, which could be challenging.

An nginx proxy may also be an option if this is too much work for Rhasspy.

**Comment by koenvervloesem** [comment]
Yes, if it's an easy addition I'd like to be able to do this without nginx. if it's too much work I can definitely try an nginx reverse proxy container and document the setup.

**Comment by koenvervloesem** [comment]
OK, so I have an nginx reverse proxy running now for Rhasspy, works like a charm. The Node-RED container on the same machine is able to connect directly to the Rhasspy container using WebSockets, while I connect to Rhasspy's web interface from other machines over HTTPS via nginx. I have also added HTTP Basic Auth in nginx.

Where in the documentation is the best location to document this setup?

**Comment by synesthesiam** [comment]
I was thinking about adding a "Recipes" section to the docs for things like this. It's not quite a tutorial, but something much more focused. What do you think?

**Comment by koenvervloesem** [comment]
Sounds fine. If you add a Recipes section, I'll add this as a recipe soon.
